react-app/src/features/profile/utils.ts

```diff
@@ -58,30 +60,44 @@ export const hasPendingRequests = (stateAccess) => {
 };
 
 // get which confirmation text to use
-export const getConfirmationModalText = (
-  selfRevokeState: StateCode | null,
-  selfWithdrawPending: boolean,
-) => {
-  if (selfRevokeState) {
+export const getConfirmationModalText = (selfRemoveRole: SelfRevokeAcess | null) => {
+  if (!selfRemoveRole) {
+    return {
+      dialogTitle: "",
+      dialogBody: "",
+      ariaLabelledBy: "",
+      dialogConfirm: "Confirm",
+    };
+  }
+  if (!selfRemoveRole.isNewUserRoleDisplay) {
     return {
       dialogTitle: "Withdraw State Access?",
       dialogBody: `This action cannot be undone. ${convertStateAbbrToFullName(
-        selfRevokeState,
+        selfRemoveRole.territory,
       )} State System Admin will be notified.`,
       ariaLabelledBy: "Self Revoke Access Modal",
+      dialogConfirm: "Confirm",
     };
   }
-  if (selfWithdrawPending) {
+  if (selfRemoveRole.status === "pending") {
     return {
       dialogTitle: "Withdraw Role Request?",
       dialogBody: "This role is still pending approval. Withdrawing it will cancel your request.",
       ariaLabelledBy: "Self Withdraw Pending Access Modal",
+      dialogConfirm: "Withdraw request",
+    };
+  }
+  if (selfRemoveRole) {
+    const approvingRole = getApprovingRole(selfRemoveRole.role);
+    const stateDisplay =
+      selfRemoveRole.territory !== "N/A"
+        ? ` ${convertStateAbbrToFullName(selfRemoveRole.territory)} `
+        : "";
+    return {
+      dialogTitle: `Withdraw ${stateDisplay}${newUserRoleMap[selfRemoveRole.role]} Access?`,
+      dialogBody: `This action cannot be undone. The ${stateDisplay}${newUserRoleMap[approvingRole]} will be notified of this change.`,
```

Comment:
`stateDisplay` might be between spaces. Would that mean that there's an unnecessary space between `The` and `${stateDisplay}`? I think you should move the space in `stateDisplay` to `dialogBody` to prevent confusion.